src/main/java/org/jabref/gui/util/DefaultTaskExecutor.java

```diff
@@ -92,19 +93,20 @@ public static void runInJavaFXThread(Runnable runnable) {
     }
 
     @Override
-    public <V> Future<V> execute(BackgroundTask<V> task) {
+    public <V> CompletableFuture<V> execute(BackgroundTask<V> task) {
         return execute(getJavaFXTask(task));
     }
 
     @Override
-    public <V> Future<V> execute(Task<V> task) {
-        executor.submit(task);
-        return task;
+    public <V> CompletableFuture<V> execute(Task<V> task) {
+        CompletableFuture<V> future = new CompletableFuture<>();
+        executor.submit((Callable<V>) task::get);
+        return future;
     }
 
     @Override
-    public <V> Future<?> schedule(BackgroundTask<V> task, long delay, TimeUnit unit) {
-        return scheduledExecutor.schedule(getJavaFXTask(task), delay, unit);
+    public <V> Future<V> schedule(BackgroundTask<V> task, long delay, TimeUnit unit) {
+        return scheduledExecutor.schedule(task::call, delay, unit);
```

Comment:
Why did you remove the getJavaFX call?

Reply:
Delegating less tasks to the JavaFX thread helps to prevent any possible GUI freezes. The JavaFX Thread should be used for UI related stuff **only**. This helps to keep the GUI interactive and improve user experience when working with JabRef. Actually there are way too many methods that delegate tasks to the UI thread. One of them should be sufficient (at most).  Everything else should be executed by another ExecutorService.

Reply:
Update: I added the method call again and managed to generify the Method so it no longer returns `Future<?>` which looks horrible to me.